File: cpp/ai-player.cpp
class AIPlayer {
public:
    static bool makeMove(std::bitset<12>& board, int roll, bool isPlayerA) {
        uint16_t mask = 1 << (roll - 1);
        
        if (isPlayerA) {
            if (board[roll - 1]) {
                board[roll - 1] = 0;
                return true;
            } else {
                for (int i = 1; i < roll; ++i) {
                    if (board[i - 1] && (roll - i <= 6)) {
                        board[i - 1] = 0;
                        return true;
                    }
                }
            }
        } else {
            if (!board[roll - 1]) {
                board[roll - 1] = 1;
                return true;
            } else {
                for (int i = 1; i < roll; ++i) {
                    if (!board[i - 1] && (roll - i <= 6)) {
                        board[i - 1] = 1;
                        return true;
                    }
                }
            }
        }
        
        return false;
    }
};

File: rust/ai-player.rs
pub struct AIPlayer;

impl AIPlayer {
    pub fn make_move(board: &mut u16, roll: u16, is_player_a: bool) -> bool {
        let mask = 1 << (roll - 1);
        
        if is_player_a {
            if *board & mask != 0 {
                *board &= !mask;
                true
            } else {
                for i in 1..roll {
                    if *board & (1 << (i - 1)) != 0 && (roll - i <= 6) {
                        *board &= !(1 << (i - 1));
                        return true;
                    }
                }
                false
            }
        } else {
            if *board & mask == 0 {
                *board |= mask;
                true
            } else {
                for i in 1..roll {
                    if *board & (1 << (i - 1)) == 0 && (roll - i <= 6) {
                        *board |= 1 << (i - 1);
                        return true;
                    }
                }
                false
            }
        }
    }
}